ethereum/contracts/interfaces/relayer/IWormholeRelayer.sol

```diff
@@ -1,376 +1,579 @@
+//  multichainSend/multichainForward
+error InsufficientMaxTransactionFee(); 
+//(maxTransactionFee, converted to target chain currency) + (receiverValue, converted to target
+//  chain currency) is greater than what your chosen relay provider allows
+error ExceedsMaximumBudget(
+  uint256 requested,
+  uint256 maximum,
+  address relayProvider,
+  uint16 chainId
+);
+
+error RelayProviderQuotedBogusAssetPrice(address relayProvider, uint16 chainId, uint256 price);
+error RelayProviderDoesNotSupportTargetChain(address relayer, uint16 chainId);
+
+//Forwards can only be requested within execution of 'receiveWormholeMessages', or when a delivery
+//  is in progress
+error NoDeliveryInProgress(); 
+// A delivery cannot occur during another delivery
+error ReentrantDelivery(address msgSender, address lockedBy);
+//A forward was requested from an address that is not the 'targetAddress' of the original delivery.
+error ForwardRequestFromWrongAddress(address msgSender, address deliveryTarget);
+
+error InvalidPayloadId(uint8 parsed, uint8 expected);
+error InvalidPayloadLength(uint256 received, uint256 expected);
+error InvalidVaaKeyType(uint8 parsed);
+
+error InvalidDeliveryVaa(string reason);
+// The delivery VAA (signed wormhole message with delivery instructions) was not emitted by the registered CoreRelayer contract
+error InvalidEmitter(bytes32 emitter, bytes32 registered, uint16 chainId);
+error VaaKeysLengthDoesNotMatchVaasLength(uint256 keys, uint256 vaas); // The VAA array has a different length than the original array of VaaKey descriptions from the source chain
+error VaaKeysDoNotMatchVaas(uint8 index); // The VAA at index 'index' does not match the 'index'-th description given on the source chain in the 'messages' field
+error RequesterNotCoreRelayer();
+
+error SendNotSufficientlyFunded(); // The container of delivery instructions (for which this current delivery was in) was not fully funded on the source chain
+error TargetChainIsNotThisChain(uint16 targetChainId); // The specified target chain is not the current chain
+error ForwardNotSufficientlyFunded(uint256 amountOfFunds, uint256 amountOfFundsNeeded); // Should never happen as this should have already been checked for
+error InvalidOverrideGasLimit(); // Invalid gas limit override was passed in to the delivery
+error InvalidOverrideReceiverValue(); // Invalid receiver value override was passed in to the delivery
+error InvalidOverrideMaximumRefund(); // Invalid maximum refund override was passed in to the delivery
+
+error InsufficientRelayerFunds(uint256 msgValue, uint256 minimum); // The relay provider didn't pass in sufficient funds (msg.value does not cover the necessary budget fees)
+
+//duplicated from utils.sol:
+error NotAnEvmAddress(bytes32);
+error OutOfBounds(uint256 offset, uint256 length);
+
+enum VaaKeyType {
+  EMITTER_SEQUENCE,
+  VAAHASH
 }
+
+/**
+ * @notice This 'VaaKey' struct identifies a wormhole message from the current transaction
```

Comment:
on second glance (this was incorrect earlier as well!) we should remove the 'from the current transaction' line! I can do this (as well as address any other of the small nits I point out) as well; just want to point it out so we don't forget